### PR TITLE
Add metrics for phase1 - client metrics

### PR DESCRIPTION
**Pull request by yontyon**
In this PR we have added a metrics framework to the bft client, including two metrics:
1. number of retransmission per client
2. the retransmission timer

Note that different clients can't use the same aggregator.

### CODE REVIEW

File: kvbc/include/ClientImp.h [hunk 1]
```diff
@@ -35,6 +35,8 @@ class ClientImp : public IClient {
                                     uint32_t* outActualReplySize,
                                     const std::string& = "") override;
 
+  virtual void setMetricsAggregator(std::shared_ptr<concordMetrics::Aggregator> aggregator) override;
```
Comment (toly-kournik) [inline kvbc/include/ClientImp.h:38]: minor: don't use 'virtual' when overriding

File: tests/simpleTest/simple_test_client.hpp [hunk 2]
```diff
@@ -74,6 +74,8 @@ class SimpleTestClient {
     ICommunication* comm = bftEngine::CommFactory::create(conf);
 
     SimpleClient* client = SimpleClient::createSimpleClient(comm, id, cp.numOfFaulty, cp.numOfSlow);
+    auto aggregator = std::make_shared<concordMetrics::Aggregator>();
```
Comment (toly-kournik) [inline tests/simpleTest/simple_test_client.hpp:77]: minor: singleliner?

File: kvbc/include/ClientImp.h (hunk 1 above)
Comment (yontyon) [reply kvbc/include/ClientImp.h:38]: All other methods in this class are defined that way.

File: tests/simpleTest/simple_test_client.hpp (hunk 2 above)
Comment (yontyon) [reply tests/simpleTest/simple_test_client.hpp:77]: I'm using the aggregator at the end of the test.

**Pull request merged**